src/lib/metrics.ts

```diff
@@ -39,376 +39,743 @@ import type { SchedulerService } from './services';
+                        appName: metric.appName,
+                    },
+                    metric.no,
+                );
+            }
+        } catch (e) {
+            logger.warn('Metrics registration failed', e);
+        }
+    });
 
-        const exceedsLimitErrorCounter = createCounter({
-            name: 'exceeds_limit_error',
-            help: 'The number of exceeds limit errors registered by this instance.',
-            labelNames: ['resource', 'limit'],
-        });
+    eventStore.on(CLIENT_REGISTER, (heartbeatEvent: ISdkHeartbeat) => {
+        if (!heartbeatEvent.sdkName || !heartbeatEvent.sdkVersion) {
+            return;
+        }
 
-        const requestOriginCounter = createCounter({
-            name: 'request_origin_counter',
-            help: 'Number of authenticated requests, including origin information.',
-            labelNames: ['type', 'method', 'source'],
-        });
+        if (flagResolver.isEnabled('extendedMetrics')) {
+            clientSdkVersionUsage.increment({
+                sdk_name: heartbeatEvent.sdkName,
+                sdk_version: heartbeatEvent.sdkVersion,
+                platform_name:
+                    heartbeatEvent.metadata?.platformName ?? 'not-set',
+                platform_version:
+                    heartbeatEvent.metadata?.platformVersion ?? 'not-set',
+                yggdrasil_version:
+                    heartbeatEvent.metadata?.yggdrasilVersion ?? 'not-set',
+                spec_version: heartbeatEvent.metadata?.specVersion ?? 'not-set',
+            });
+        } else {
+            clientSdkVersionUsage.increment({
+                sdk_name: heartbeatEvent.sdkName,
+                sdk_version: heartbeatEvent.sdkVersion,
+                platform_name: 'not-set',
+                platform_version: 'not-set',
+                yggdrasil_version: 'not-set',
+                spec_version: 'not-set',
+            });
+        }
+    });
 
-        const resourceLimit = createGauge({
-            name: 'resource_limit',
-            help: 'The maximum number of resources allowed.',
-            labelNames: ['resource'],
-        });
+    eventStore.on(PROJECT_ENVIRONMENT_REMOVED, ({ project }) => {
+        projectEnvironmentsDisabled.increment({ project_id: project });
+    });
 
-        const addonEventsHandledCounter = createCounter({
-            name: 'addon_events_handled',
-            help: 'Events handled by addons and the result.',
-            labelNames: ['result', 'destination'],
-        });
+    eventBus.on(events.ADDON_EVENTS_HANDLED, ({ result, destination }) => {
+        addonEventsHandledCounter.increment({ result, destination });
+    });
 
-        async function collectStaticCounters() {
+    return {
+        collectDbMetrics: dbMetrics.refreshDbMetrics,
+        collectStaticCounters: async () => {
             try {
-                dbMetrics.refreshDbMetrics();
-
-                const stats = await instanceStatsService.getStats();
```

Comment:
getStats was triggering many queries in parallel, some of which were not used in prometheus.